### ISSUE TITLE
Adaptive(?) Error Messages

### ISSUE THREAD

**Issue by wgledbetter** [AlabamaASRL/asset_asrl#35]
Any time an error is thrown as a result of evaluating a conditional, the arguments to the conditional should be embedded in the error message.

**Comment by jbpezent** [comment]
@wgledbetter Agreed, line numbers are already handled when the exception is thrown to python, but there is alot more information that we could and should provide in the error messages, regarding IO sizes, to help diagnose issues.
